core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -281,8 +281,9 @@ public void registerHighlighter(String key, Class<? extends Highlighter> clazz)
         highlighters.registerExtension(key, clazz);
     }
 
-    public void registerSuggester(String key, Class<? extends Suggester> suggester) {
+    public void registerSuggester(String key, Class<? extends Suggester> suggester, NamedWriteable<? extends SuggestionBuilder> builder) {
```

Comment:
There is a `getBuilderPrototype()` method each suggester needs to provide which should return the appropriate builder prototype for each suggester. You could register the Suggester itself and then get its class and its builder prototype.